### ISSUE TITLE
Etch-a-Sketch: Clarification of the instructions in the extra credit section

### ISSUE THREAD

**Issue by FireHeart3198** [TheOdinProject/curriculum#27500]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

For the extra credit section, instruction no. 2 which states, 

>Additionally, implement a progressive darkening effect where each interaction adds 10% more black or color to the square. The objective is to achieve a completely black square only after ten interactions.

could potentially be misinterpreted as darkening the 'universal' colour for all squares to black in 10 steps instead of a shading effect which the instruction seems to intend. A potential revision could be: 

>Additionally, implement a progressive darkening effect where each interaction **with the same square** adds 10% more black or color to **that** square. The objective is to achieve a completely black square only after ten interactions.

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-etch-a-sketch

### (Optional) Discord Name

fireheart6464

**Comment by MaoShizhong** [comment]
Thank you for opening this issue, @FireHeart3198 

I'm not so sure this is something that really needs changing - I personally think the current text is clear enough that the darkening effect should affect `the square` that the interaction occurs on, and that `a square` thus becomes black after ten such interactions with it. That being said, it's not something I feel really strongly about *not* changing/that your suggestion is "worse", only that I'm not sure it really changes a great deal in either direction.

Was there a situation/some situations you observed in the server with people getting confused that led to this suggestion? I haven't recently been interacting with much help regarding Etch-a-sketch so haven't bumped into any of these myself, hence haven't considered any need yet.

**Comment by FireHeart3198** [comment]
I personally had interpreted it as an overall darkening effect, and so targeted a global variable that was changed with each interaction instead of checking what level of shading the square had. I had not realised this until reading a recent conversation in the server between someone who also interpreted it the same way and Damon who corrected us towards the shading effect, which is why I thought to raise the issue.

It is a very minor thing that might have only confused a small number and of course it's an extra credit assignment so not changing it wouldn't impact the overall project but I'd thought I'd raise the issue nonetheless :)

**Comment by MaoShizhong** [comment]
@FireHeart3198 I see! 

I've had a look back through the server to try and find if there were more cases with similar confusion but couldn't find any, and don't recall this specific confusion being very common prior to that when I'd be involved with that more. So I don't think this change feels *necessary* right now, hence I will not do anything with it yet.

I will leave the issue open for a few more days though and see if anyone else (maintainer or not) has any input/insight. If you happen to notice any similar kind of confusion in the server yourself then by all means throw a link to the conversation(s) here!

**Comment by cakegod** [comment]
I am with Mao in this one; I haven't seen any Odinite having issues understanding that the darkening should be applied to individual cells.

One thing that we can add, though, is hints on how to implement this feature. It's something with which some Odinites tend to have difficulty.

**Comment by FireHeart3198** [comment]
Oh I do agree with that! Out of all the tasks the darkening effect seems like a really good way to get practice checking the state of an element but it does seem to be especially difficult for people to implement so a slight nudge would be great. And it would essentially nix any chance of misunderstanding too.

**Comment by cakegod** [comment]
A small mention that `rgb` takes a fourth (optional) value to control transparency should be good enough.

**Comment by MaoShizhong** [comment]
> A small mention that `rgb` takes a fourth (optional) value to control transparency should be good enough.

@cakegod Would `opacity` be a simpler option to mention? Since with `rgba`, I thought you'd have to split by `,` before parseFloat-ing to get the current alpha (unless there's a much simpler way I'm just not aware of, which is definitely very possible XD).
With opacity, people can just start at 0 and add 0.1 each time (no need to split, only parseFloat/Number it). If all that's beneficial would be a tiny mention like "the XXXXX property might be useful here" then my gut tells me that's a little simpler.

**Comment by cakegod** [comment]
@MaoShizhong  I agree; it's easier to use opacity. The first optional assignment mentions `rgb`, but you made a fair point about the complexity of obtaining the alpha.  :stuck_out_tongue_winking_eye:

If no one is willing, I can make the PR.

**Comment by MaoShizhong** [comment]
@cakegod if you can add the opacity note, that should be good.

You may also want to slightly reword the current step about "end result achieve a fully black square after 10 interactions".
It currently says "at a little more black or colour", and achieving full black when in rainbow mode would be a bit much. 

Either keep it as black but remove the colour mention from it, or keep the colour mention but adjust the wording of the expected outcome after 10 interactions accordingly. I'll trust your judgement on that one.

**Comment by cakegod** [comment]
@MaoShizhong I think we should just remove the colour part. 

Anyway, assign me this please.
